### ISSUE TITLE
Automatic snapshot schedule issues

### ISSUE THREAD

**Issue by okrause** [NetApp/terraform-provider-netapp-gcp#10]
Bug report on c825588:

Automatic snapshot schedules don't get applied correctly. Testfile:
[volume.tf.txt](https://github.com/NetApp/terraform-provider-netapp-gcp/files/4831022/volume.tf.txt)

Doing terraform apply, volume doesn't exist:
```bash
terraform/CVS/volume terraform apply  

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # netapp-gcp_volume.gcp-volume will be created
  + resource "netapp-gcp_volume" "gcp-volume" {
      + id             = (known after apply)
      + name           = "okterraform"
      + network        = "ncv-vpc"
      + protocol_types = [
          + "NFSv3",
        ]
      + region         = "europe-west3"
      + service_level  = "premium"
      + size           = 1024
      + volume_path    = (known after apply)

      + export_policy {
          + rule {
              + access          = "ReadWrite"
              + allowed_clients = "0.0.0.0/0"

              + nfsv3 {
                  + checked = true
                }

              + nfsv4 {
                  + checked = false
                }
            }
        }

      + mount_points {
          + export        = (known after apply)
          + protocol_type = (known after apply)
          + server        = (known after apply)
        }

      + snapshot_policy {
          + enabled = true

          + daily_schedule {
              + hour              = 23
              + minute            = 2
              + snapshots_to_keep = 14
            }

          + hourly_schedule {
              + minute            = 1
              + snapshots_to_keep = 48
            }

          + monthly_schedule {
              + days_of_month     = "6"
              + hour              = 2
              + minute            = 4
              + snapshots_to_keep = 6
            }

          + weekly_schedule {
              + day               = "Monday"
              + hour              = 1
              + minute            = 3
              + snapshots_to_keep = 4
            }
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

netapp-gcp_volume.gcp-volume: Creating...
netapp-gcp_volume.gcp-volume: Still creating... [10s elapsed]
netapp-gcp_volume.gcp-volume: Creation complete after 11s [id=f036b7bf-1287-6fca-d5ea-e0e88287394c]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```

Looking at the UI, I get the following setting:
Hourly: 48 snaps, minute 1. Good
Daily: 14 snaps, Hour 23, minute 2. Good
Weekly: 4 snaps, Monday, Hour 1, Minute 3. Good.
Monthly: 0 snaps, Days 1, Hour 0, Minute 0. **BAD**

Setting monthly didn't work.

Additionally, terraform state seems to be incorrect. A subsequent terraform apply with the same file yields:
```bash
Terraform/CVS/volume terraform apply
netapp-gcp_volume.gcp-volume: Refreshing state... [id=f036b7bf-1287-6fca-d5ea-e0e88287394c]

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # netapp-gcp_volume.gcp-volume will be updated in-place
  ~ resource "netapp-gcp_volume" "gcp-volume" {
        id             = "f036b7bf-1287-6fca-d5ea-e0e88287394c"
        name           = "okterraform"
        network        = "ncv-vpc"
        protocol_types = [
            "NFSv3",
        ]
        region         = "europe-west3"
        service_level  = "premium"
        size           = 1024
        volume_path    = "stoic-affectionate-heyrovsky"

        export_policy {
            rule {
                access          = "ReadWrite"
                allowed_clients = "0.0.0.0/0"

                nfsv3 {
                    checked = true
                }

                nfsv4 {
                    checked = false
                }
            }
        }

        mount_points {
            export        = "/stoic-affectionate-heyrovsky"
            protocol_type = "NFSv3"
            server        = "10.194.0.36"
        }

      ~ snapshot_policy {
            enabled = true

            daily_schedule {
                hour              = 23
                minute            = 2
                snapshots_to_keep = 14
            }

          ~ hourly_schedule {
                minute            = 1
              ~ snapshots_to_keep = 0 -> 48
            }

          ~ monthly_schedule {
              ~ days_of_month     = "1" -> "6"
              ~ hour              = 0 -> 2
              ~ minute            = 0 -> 4
              ~ snapshots_to_keep = 0 -> 6
            }

            weekly_schedule {
                day               = "Monday"
                hour              = 1
                minute            = 3
                snapshots_to_keep = 4
            }
        }
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

As you can see, state for hourly, monthly and weekly seems to be inconsistent with what was set by the first apply and by the state the UI is showing.

**Comment by lonico** [comment]
We identified 2 issues in the code.  A PR is under internal review.

**Comment by okrause** [comment]
Issue got resolved with ac5f53ee79fc57d914d29df2f6110d16d1f0bdf9
Closing
